content/custom_resources/lwrp_to_custom_resources.md

```diff
@@ -0,0 +1,123 @@
++++
+title = "Migrating from LWRPs to Custom Resources"
+gh_repo = "chef-web-docs"
+
+[menu]
+  [menu.infra]
+    title = "Migrating from LWRPs"
+    identifier = "chef_infra/cookbook_reference/custom_resources/lwrp"
+    parent = "chef_infra/cookbook_reference/custom_resources"
+    weight = 50
++++
+
+## Overview
+
+It is no longer recommended to write resources in the __Light Weight Resource Provider (LWRP)__ format.
+
+This guide describes how to migrate from an existing LWRP to a Custom Resource.
+
+If you are creating a Custom Resource from scratch please see the [Custom Resource Getting Started Guide](getting-started.md) instead.
+
+## Convert files to Custom Resources Layout
+
+LWRPs are a pair of library files one provider, one resource.
+
+```text
+└── libraries
+    ├── provider_rvm_ruby.rb
+    └── resource_rvm_ruby.rb
+```
+
+These files are merged into one, and moved into the resources directory.
+
+```text
+└── resources
+    └── rvm_ruby.rb
+```
+
+## Drop LWRP classes
+
+LWRPs required telling Chef, that they were not regular library files, but Providers and Resources. This is now taken care of for us.
```

Comment:
Recommend similar to: LWRPs used classes to separate Provider and Resource behaviors, but custom resources do not need this distinction. This means that we remove the class definitions in their entirety, as shown in the following example: